File: .sdk-repos/orchestration-cluster-api-python/examples/message_signal.py
# Compilable usage examples for message and signal operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    MessageCorrelationRequest,
    MessagePublicationRequest,
    SignalBroadcastRequest,
)

# region CorrelateMessage
def correlate_message_example() -> None:
    client = CamundaClient()

    result = client.correlate_message(
        data=MessageCorrelationRequest(
            name="payment-received",
            correlation_key="order-12345",
        )
    )

    print(f"Message key: {result.message_key}")
# endregion CorrelateMessage


# region PublishMessage
def publish_message_example() -> None:
    client = CamundaClient()

    result = client.publish_message(
        data=MessagePublicationRequest(
            name="order-created",
            correlation_key="order-12345",
            time_to_live=60000,
        )
    )

    print(f"Message key: {result.message_key}")
# endregion PublishMessage


# region BroadcastSignal
def broadcast_signal_example() -> None:
    client = CamundaClient()

    result = client.broadcast_signal(
        data=SignalBroadcastRequest(
            signal_name="order-cancelled",
        )
    )

    print(f"Signal key: {result.signal_key}")
# endregion BroadcastSignal

File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/MessageSignal.cs
// Compilable usage examples for message and signal operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class MessageSignalExamples
{
    #region CorrelateMessage
    // <CorrelateMessage>
    public static async Task CorrelateMessageExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.CorrelateMessageAsync(new MessageCorrelationRequest
        {
            Name = "paymentReceived",
            CorrelationKey = "order-123",
        });

        Console.WriteLine($"Message key: {result.MessageKey}");
    }
    // </CorrelateMessage>
    #endregion CorrelateMessage

    #region PublishMessage

    // <PublishMessage>
    public static async Task PublishMessageExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.PublishMessageAsync(new MessagePublicationRequest
        {
            Name = "paymentReceived",
            CorrelationKey = "order-123",
            TimeToLive = 60000,
        });

        Console.WriteLine($"Message key: {result.MessageKey}");
    }
    // </PublishMessage>
    #endregion PublishMessage

    #region BroadcastSignal

    // <BroadcastSignal>
    public static async Task BroadcastSignalExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.BroadcastSignalAsync(new SignalBroadcastRequest
        {
            SignalName = "orderCancelled",
        });

        Console.WriteLine($"Signal key: {result.SignalKey}");
    }
    // </BroadcastSignal>
    #endregion BroadcastSignal

    #region SearchMessageSubscriptions

    // <SearchMessageSubscriptions>
    public static async Task SearchMessageSubscriptionsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchMessageSubscriptionsAsync(
            new MessageSubscriptionSearchQuery());

        foreach (var sub in result.Items)
        {
            Console.WriteLine($"Subscription: {sub.MessageName}");
        }
    }
    // </SearchMessageSubscriptions>
    #endregion SearchMessageSubscriptions

    #region SearchCorrelatedMessageSubscriptions

    // <SearchCorrelatedMessageSubscriptions>
    public static async Task SearchCorrelatedMessageSubscriptionsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchCorrelatedMessageSubscriptionsAsync(
            new CorrelatedMessageSubscriptionSearchQuery());

        foreach (var sub in result.Items)
        {
            Console.WriteLine($"Correlated subscription: {sub.MessageName}");
        }
    }
    // </SearchCorrelatedMessageSubscriptions>
    #endregion SearchCorrelatedMessageSubscriptions
}
